pim-tool-back/src/main/java/pilotproject/Project_Management_ELCA/service/impl/ProjectServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+package pilotproject.Project_Management_ELCA.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import pilotproject.Project_Management_ELCA.model.dto.ProjectDto;
+import pilotproject.Project_Management_ELCA.model.entity.Project;
+import pilotproject.Project_Management_ELCA.model.exception.ProjectNumberExistedException;
+import pilotproject.Project_Management_ELCA.repository.GroupRepository;
+import pilotproject.Project_Management_ELCA.repository.ProjectRepository;
+import pilotproject.Project_Management_ELCA.service.ProjectService;
+import pilotproject.Project_Management_ELCA.util.ApplicationMapper;
+
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Service
+//@Transactional
+public class ProjectServiceImpl implements ProjectService {
+
+    @Autowired
+    private ApplicationMapper mapper;
+
+    private final ProjectRepository projectRepository;
+
+    @Autowired
+    private GroupRepository groupRepository;
+
+    @Autowired
+    public ProjectServiceImpl(ProjectRepository projectRepository) {
+        this.projectRepository = projectRepository;
+    }
+
+    @Override
+    public ProjectDto addProject(ProjectDto dto) throws ProjectNumberExistedException {
+        Project project = mapper.projectDtoToProjectEntity(dto);
+        project.setGroup(groupRepository.findById(dto.getGroupId()).orElseThrow());
+        try {
+            return mapper.projectToProjectDto(projectRepository.save(project));
+        } catch (DataIntegrityViolationException e) {
+            throw new ProjectNumberExistedException("The project number already existed. Please select a different project number");
```

Comment:
DataIntegrityViolationException can mean anything related to data integrity. How can you make sure it is because of project number already existed?